### DISCUSSION TITLE
discount amount is not shown in checkout

### DISCUSSION THREAD

**Discussion by ghost** [saleor/saleor#12499] [Q&A]
there is a fixed discount present on certain products

i have added multiple products in my checkout which contain some discounted products 
i need to show the total discount on summary page before creating an order
currently, checkout.discount.amount is showing as 0, which ideally should be the total discount amount 
 
so now to calculate the total discount amount, we are looping on all line items and adding the difference of (undiscountedTotalPrice-totalPrice.net.amount) 
whereas in order detail i can directly get the total discount amount 

so, do we have any existing support for the total discount, or do we have this planned in the near future?

thanks

**Comment by korycins** [comment]
Hey @sameer-setia 

> whereas in order detail i can directly get the total discount amount

How do you do that? I don’t think that `order.discount` or `order.discounts` include the sale discount. 
The `Order` type has an `undiscountedTotal` field. It seems like we could also introduce this field to the `Checkout` type. Would this solve your problem?
